examples/suit_update/README.md

```diff
@@ -369,10 +369,10 @@ see 6 pairs of messages indicating where (filepath) the file was published and
 the corresponding coap resource URI
 
     ...
-    published "/home/francisco/workspace/RIOT/examples/suit_update/bin/samr21-xpro/suit_update-riot.suitv4_signed.1557135946.bin"
-           as "coap://[2001:db8::1]/fw/samr21-xpro/suit_update-riot.suitv4_signed.1557135946.bin"
-    published "/home/francisco/workspace/RIOT/examples/suit_update/bin/samr21-xpro/suit_update-riot.suitv4_signed.latest.bin"
-           as "coap://[2001:db8::1]/fw/samr21-xpro/suit_update-riot.suitv4_signed.latest.bin"
+    published "/home/francisco/workspace/RIOT/examples/suit_update/bin/samr21-xpro/suit_update-riot.suitv3_signed.1557135946.bin"
+           as "coap://[2001:db8::1]/fw/samr21-xpro/suit_update-riot.suitv3_signed.1557135946.bin"
+    published "/home/francisco/workspace/RIOT/examples/suit_update/bin/samr21-xpro/suit_update-riot.suitv3_signed.latest.bin"
+           as "coap://[2001:db8::1]/fw/samr21-xpro/suit_update-riot.suitv3_signed.latest.bin"
     ...
 
 ### Notify an update to the device
```

Comment:
Change `fe80::2%riot0` by the EUI64 address, `RIOT` will answer with the EUI64 address and this will confuse aiocoap:

```
WARNING:coap:Received Type.ACK from <UDP6EndpointAddress [fe80::be:74ff:fec0:2fb9%riot0] with local address>, but could not match it to a running exchange.
```

It will lead to it re-transmitting the manifest multiple times

```
riotboot_flashwrite: initializing update to target slot 0
suit: received URL: "coap://[2001:db8::1]/fw/usb-kw41z/suit_update-riot.suitv3_signed.latest.bin"
Fetching firmware |██                       |   7%suit: received URL: "coap://[2001:db8::1]/fw/usb-kw41z/suit_update-riot.suitv3_signed.latest.bin"
Fetching firmware |██████████████████       |  72%suit: received URL: "coap://[2001:db8::1]/fw/usb-kw41z/suit_update-riot.suitv3_signed.latest.bin"
Fetching firmware |█████████████████████████| 100%
Verifying image digest
```

Using EUI64 it works like a charm.

Reply:
Just to be sure, is this an issue that we should fix here?

Reply:
No, this has to do with how `source` selection is performed, when source addresses have the same amount of points (according to source selection rules) `RIOT` will always select the EUI64 address for link local addresses when answering pings or whatever (I think its because its the first address and EUI64 is always the first added address).

There is no way of knowing which addresses of a specific interface a message came in. We are complying with the RFC.